src/ophyd_async/epics/adcore/_hdf_writer.py

```diff
@@ -51,16 +51,15 @@ def __init__(
     async def open(self, multiplier: int = 1) -> Dict[str, DataKey]:
         self._file = None
         info = self._path_provider(device_name=self.hdf.name)
-        file_path = info.directory_path
         await asyncio.gather(
             self.hdf.num_extra_dims.set(0),
             self.hdf.lazy_open.set(True),
             self.hdf.swmr_mode.set(True),
+            self.hdf.create_dir_depth.set(info.create_dir_depth),
```

Comment:
The gather will fire them off but there's no guarantee in the order they will be received. Better to `await self.hdf.create_dir_depth` before doing the `await asyncio.gather` of all the other signals